src/app/admin/_components/faq/List.js

```diff
@@ -0,0 +1,131 @@
+'use client';
+
+import { Switch } from '@mui/material';
+import {
+  Datagrid,
+  List,
+  TextField,
+  WrapperField,
+  useNotify,
+  useRecordContext,
+  useRefresh,
+  useUpdate,
+} from 'react-admin';
+
+import { FAQ_PRIORITY_CHANGE_STEP } from '@/lib/consts';
+import { MAX_ACTIVE_FAQS, MIN_ACTIVE_FAQS } from './consts';
+import { LinkTextField } from '../LinkTextField';
+import { UpDownArrowMenu } from '../UpDownArrowMenu';
+import { useActiveFaqs } from './hooks';
+
+function ChangePriorityButtons() {
+  const notify = useNotify();
+  const record = useRecordContext();
+  const [update, { isLoading, error }] = useUpdate('faq');
+  const refresh = useRefresh();
+
+  const handleError = queryError => {
+    notify(`Неможливо оновити дані\nПомилка: ${queryError.message}`, { type: 'error' });
+  };
+
+  const handleChange = diff => () => {
+    const isDiffPositive = diff > 0;
+    update(
+      'Faq',
+      { id: record.id, data: { priority: record.priority + diff }, previousData: record },
+      {
+        onSuccess: () => {
+          notify(`Приорітет ${isDiffPositive ? 'підвищено' : 'знижено'}`);
+          refresh();
+        },
+        onError: handleError,
+      },
+    );
+  };
+
+  if (error) {
+    return handleError(error);
+  }
+
+  return (
+    <UpDownArrowMenu
+      onIncrease={handleChange(FAQ_PRIORITY_CHANGE_STEP)}
+      onDecrease={handleChange(-FAQ_PRIORITY_CHANGE_STEP)}
+      disabled={isLoading}
+    >
+      {record.priority ? <TextField source="priority" /> : '-'}
+    </UpDownArrowMenu>
+  );
+}
+
+function IsActiveSwitch() {
+  const notify = useNotify();
+  const record = useRecordContext();
+  const [update, { isLoading, error }] = useUpdate('faq');
+  const { total: activeFaqsCount, isLoading: isLoadingActiveFaqs, error: faqsLoadingError } = useActiveFaqs();
+  const refresh = useRefresh();
+
+  if (isLoadingActiveFaqs || faqsLoadingError) {
+    return null;
+  }
+
+  const handleError = queryError => {
+    notify(`Неможливо оновити дані\nПомилка: ${queryError.message}`, { type: 'error' });
+  };
```

Comment:
Thanks for comment! Unfortunately its impossible to do so because we can't move `notify` out of component(`useNotify` cannot be called at the top level).

Reply:
You are right. Thanks for the explanation.